src/Microsoft.DotNet.Build.Tasks.Feed/FeedActionTask.cs

```diff
@@ -0,0 +1,136 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+using MSBuild = Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.DotNet.Build.Tasks.Feed
+{
+
+    public class FeedActionTask : MSBuild.Task
+    {
+
+        [Required]
+        public string Task { get; set; }
+        
+        public string Type { get; set; }
+        
+        public string FeedAccountName { get; set; }
+        
+        public string FeedAccountKey { get; set; }
+
+        public bool Debug { get; set; }
+
+        public ITaskItem[] ItemsToPublish { get; set; }
+
+        public string RelativePath { get; set; }
+
+        public bool PublishFlatContainer { get; set; }
+
+        public bool Overwrite { get; set; }
+
+        public ITaskItem[] EnableFeeds { get; set; }
+
+        public ITaskItem[] DisableFeeds { get; set; }
+        public string RepoRoot { get; set; }
+
+        public override bool Execute()
+        {
+            return ExecuteAsync().GetAwaiter().GetResult();
+        }
+
+        public async Task<bool> ExecuteAsync()
+        {
+            if(Debug)
+                Debugger.Launch();
+            try
+            {
+                switch (Task)
+                {
+                    case "Push":
+                        Log.LogMessage(MessageImportance.High, "Performing feed push...");
+                        if (ItemsToPublish == null)
+                        {
+                            Log.LogError($"No items to push. Please check ItemGroup ItemsToPublish.");
+                        }
+                        if (string.IsNullOrEmpty(RelativePath))
+                        {
+                            Log.LogWarning($"No relative path. Items are pushed to root of container.");
+                        }
+                        if (Type == "Blob") {
+                            //FeedAccountName is of the form - https://account.blob.core.windows.net/container/
+                            string removeHeader = FeedAccountName.Substring(FeedAccountName.IndexOf("https://")+8);
```

Comment:
This parsing seems a little too magical and making heavy assumptions about the FeedAccountName input string.

Reply:
fixed.